src/Development/IDE/Core/Rules.hs

```diff
@@ -504,6 +504,21 @@ getDependenciesRule =
         let mbFingerprints = map (fingerprintString . fromNormalizedFilePath) allFiles <$ optShakeFiles opts
         return (fingerprintToBS . fingerprintFingerprints <$> mbFingerprints, ([], transitiveDeps depInfo file))
 
+getHieAstsRule :: Rules ()
+getHieAstsRule =
+    define $ \GetHieAst f -> do
```

Comment:
How come this rule isn't used anywhere?

Reply:
It's meant to be used in `hls` code for now. 

I have code in the `hls` branch that replaces all usages of `GetSpanInfo` with this. However, since we can't generate HIE information for 8.4, so I've not included that in this patch.

One thing we can do is use it to do is to produce scope-aware local completions. I believe @isovector is working on generating one crucial requirement for this as part of his tactics PR(the interval map telling us which names are in scope at a point). The other part is #785, after which it should be just a matter of tying the two things together. We can wait for this to land before merging this patch.

Reply:
I'm happy to drop 8.4 compatibility in the next ghcide release.

Reply:
But that said, why not change `getDefinition` and `getTypeDefinition` to use `GetHieAsts` in this PR?